tests/components/dyson/test_sensor.py

```diff
@@ -269,7 +303,47 @@ async def test_purecool_component_setup_only_once(devices, login, hass):
     config = _get_config()
     await async_setup_component(hass, dyson_parent.DOMAIN, config)
     await hass.async_block_till_done()
-    discovery.load_platform(hass, "sensor", dyson_parent.DOMAIN, {}, config)
+
+    assert len(hass.data[dyson.DYSON_SENSOR_DEVICES]) == 4
```

Comment:
Assert the number of states in the state machine under the sensor domain instead.

Reply:
Do you mind if I back out of changing this test altogether, and just change the number to 4 so that that particular test passes?

I'm not quite sure I understand how this part of the test is meant to work, or what it's exactly meant to test :/

Reply:
Yeah, that's ok.